File: app/src/main/java/com/nuhkoca/myapplication/di/qualifier/DataBinding.java
package com.nuhkoca.myapplication.di.qualifier;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.inject.Scope;

/**
 * A scope for DataBinding classes.
 *
 * @author nuhkoca
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
public @interface DataBinding {
}

File: app/src/main/java/com/nuhkoca/myapplication/di/scope/DataBinding.kt
package com.nuhkoca.myapplication.di.scope

import javax.inject.Scope

/**
 * A scope for DataBinding classes.
 *
 * @author nuhkoca
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.CLASS,
    AnnotationTarget.FILE
)
annotation class DataBinding
